### PR TITLE
Proposal: Enable refinements to `#method_missing`

### PR DESCRIPTION
**Pull request by osyo-manga**

### CODE REVIEW

File: spec/ruby/core/module/refine_spec.rb [hunk 1]
```diff
@@ -560,6 +560,45 @@ def foo; end
...
+        -> {
+          Module.new do
+            using refinement
+            klass.new.hoge
+          end
+        }.should raise_error(NameError, /undefined method `hoge'/)
+      end
+    end
+
+    ruby_version_is "2.6" do
```
[the remark this answers is not in the document]
Comment (osyo-manga) [reply spec/ruby/core/module/refine_spec.rb]: Thanks for comment.
Probably it is judged with 2.6 or higher.
https://github.com/ruby/ruby/blob/30c4b648cace4b00cc584f20aec8a62341c05f43/spec/mspec/lib/mspec/guards/version.rb#L26-L32

[the remark this answers is not in the document]
Comment (eregon) [reply spec/ruby/core/module/refine_spec.rb]: Right, `ruby_version_is "2.6"` is correct.
It's a bit confusing, I know.
I'm thinking to let `"2.6"...""` work at least.

**Comment by k0kubun** [comment]
It seems to have a conflict now. Could you rebase this from master?

**Comment by osyo-manga** [comment]
Close because it was rejected.
https://bugs.ruby-lang.org/issues/15374
Thank you :)

**Pull request closed without merging**